src/ExperimentalCycles.mo

```diff
@@ -1,51 +1,152 @@
+///     };
+///
+///     func operation(): async() {
+///         Debug.print("Operation balance: " # debug_show(Cycles.balance()));
+///         Debug.print("Operation available: " # debug_show(Cycles.available()));
+///         let obtained = Cycles.accept(10_000_000);
+///         Debug.print("Operation obtained: " # debug_show(obtained)); // => 10_000_000
+///         Debug.print("Operation balance: " # debug_show(Cycles.balance())); // increased by 10_000_000
+///         Debug.print("Operation available: " # debug_show(Cycles.available())); // decreased by 10_000_000
+///     }
+/// }
+/// ```
 import Prim "mo:⛔";
 module {
 
   /// Returns the actor's current balance of cycles as `amount`.
+  ///
+  /// Example for use on the IC:
+  /// ```motoko
+  /// import Cycles "mo:base/ExperimentalCycles";
+  /// import Debug "mo:base/Debug";
+  ///
+  /// actor {
+  ///     public func main(): async() {
+  ///         let balance = Cycles.balance();
+  ///         Debug.print("Balance: " # debug_show(balance));
+  ///     }
+  /// }
+  /// ```
   public let balance : () -> (amount : Nat) = Prim.cyclesBalance;
 
   /// Returns the currently available `amount` of cycles.
   /// The amount available is the amount received in the current call,
   /// minus the cumulative amount `accept`ed by this call.
-  /// On exit from the current shared function or async expression via `return` or `throw`
-  /// any remaining available amount is automatically
-  /// refunded to the caller/context.
+  /// On exit from the current shared function or async expression via `return` or `throw`,
+  /// any remaining available amount is automatically refunded to the caller/context.
+  ///
+  /// Example for use on the IC:
+  /// ```motoko
+  /// import Cycles "mo:base/ExperimentalCycles";
+  /// import Debug "mo:base/Debug";
+  ///
+  /// actor {
+  ///     public func main(): async() {
+  ///         let available = Cycles.available();
+  ///         Debug.print("Available: " # debug_show(available));
+  ///     }
+  /// }
+  /// ```
   public let available : () -> (amount : Nat) = Prim.cyclesAvailable;
 
   /// Transfers up to `amount` from `available()` to `balance()`.
   /// Returns the amount actually transferred, which may be less than
   /// requested, for example, if less is available, or if canister balance limits are reached.
+  ///
+  /// Example for use on the IC:
+  /// ```motoko
+  /// import Cycles "mo:base/ExperimentalCycles";
+  /// import Debug "mo:base/Debug";
+  ///
+  /// actor {
+  ///     public func main(): async() {
+  ///         Cycles.add(15_000_000);
+  ///         await operation(); // accepts 10_000_000 cycles
+  ///     };
+  ///
+  ///     func operation(): async() {
+  ///         let obtained = Cycles.accept(10_000_000);
+  ///         Debug.print("Obtained: " # debug_show(obtained)); // => 10_000_000
+  ///     }
+  /// }
+  /// ```
```

Comment:
If I'm understanding correctly, is this transferring cycles from one canister to itself?

Reply:
Yes, I was unsure whether this makes sense. I chose it because of simplicity of the example to avoid a second actor (as they would be placed in separate files, and the example would consist of two files). Maybe I should introduce a second actor. But then I would need two files in the examples, for the second actor or a module with an actor class.
Or maybe there is another way.

Reply:
Update: I added a comment that this is a simplified example (self-transfer). Do you think this is okay? I am happy to reopen the PR and adjust it.

Reply:
Yes I think it's okay!